cmd/snap/cmd_run.go

```diff
@@ -303,9 +303,18 @@ func createUserDataDirs(info *snap.Info) error {
 	}
 
 	// see snapenv.User
-	userData := info.UserDataDir(usr.HomeDir)
-	commonUserData := info.UserCommonDataDir(usr.HomeDir)
-	for _, d := range []string{userData, commonUserData} {
+	instanceUserData := info.UserDataDir(usr.HomeDir)
+	instanceCommonUserData := info.UserCommonDataDir(usr.HomeDir)
+	createDirs := []string{instanceUserData, instanceCommonUserData}
+	if info.InstanceKey != "" {
+		// parallel instance snaps get additional mapping in their mount
+		// namespace, namely /home/joe/snap/foo_bar ->
+		// /home/joe/snap/foo, make sure that the mount point exists and
+		// is owned by the user
+		snapUserDir := filepath.Join(usr.HomeDir, dirs.UserHomeSnapDir, info.SnapName())
```

Comment:
Above we have the pattern of calling methods like info.SomeUserSpecificLocation(usr.HomeDir), is there a reason not to do it here as well?

Reply:
Yeah, this feels a bit inconsistent at first look at least.

Reply:
Let me add a helper then. IIRC there's one more place that may need an update as well.

Reply:
This is done now.